### PR TITLE
Secrets: Add unified secrets table to reencryption

### PR DESCRIPTION
**Pull request by guicaulada**
**What this PR does / why we need it**:

This PR adds the new unified secrets table to the rencryption logic.

**Which issue(s) this PR fixes**:

There is no issue for this, but this fixes the problem of the new secrets table not being reencrypted by the grafana-cli.

**Special notes for your reviewer**:

### CODE REVIEW

**Review by joanlopez — CHANGES_REQUESTED** [review]
Hi @guicaulada,

Thanks for working on this. However, after a quick review, looks like rows in `secrets` table contain an `updated` field but the `b64secret` implementation present on `secretsmigrations` commands does not update it.

Could you update that implementation to conditionally update the `updated` column, please? 

You can look at the example from `simpleSecret` as a reference for the SQL query and add a boolean field into `b64Secret` to determine if the given secret type (database rows) have an `updated` field or not.

What do you think? Thanks!

**Comment by guicaulada** [comment]
> Hi @guicaulada,
> 
> Thanks for working on this. However, after a quick review, looks like rows in `secrets` table contain an `updated` field but the `b64secret` implementation present on `secretsmigrations` commands does not update it.
> 
> Could you update that implementation to conditionally update the `updated` column, please?
> 
> You can look at the example from `simpleSecret` as a reference for the SQL query and add a boolean field into `b64Secret` to determine if the given secret type (database rows) have an `updated` field or not.
> 
> What do you think? Thanks!

Done on 536c042

File: pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go [hunk 1]
```diff
@@ -256,9 +261,10 @@ func ReEncryptSecrets(_ utils.CommandLine, runner runner.Runner) error {
 		reencrypt(*manager.SecretsService, *xorm.Session)
 	}{
 		simpleSecret{tableName: "dashboard_snapshot", columnName: "dashboard_encrypted"},
-		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_access_token"}},
-		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_refresh_token"}},
-		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_token_type"}},
+		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_access_token"}, false},
+		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_refresh_token"}, false},
+		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_token_type"}, false},
+		b64Secret{simpleSecret{tableName: "secrets", columnName: "value"}, true},
```
Comment (joanlopez) [inline pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go:267]:
```suggestion
		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_access_token"}},
		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_refresh_token"}},
		b64Secret{simpleSecret{tableName: "user_auth", columnName: "o_auth_token_type"}},
		b64Secret{simpleSecret{tableName: "secrets", columnName: "value"}, hasUpdatedColumn: true},
```

**Review by joanlopez — APPROVED** [review]
Great! Nice job! 🚀 

Only left a nitpick suggestion, in favor of default values and to clarify what the boolean values does mean. 
Anyway, it looks good to me, so feel free to merge it as-is! ✅ 

Thanks!

File: pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go (hunk 1 above)
Comment (joanlopez) [reply pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go:267]: Mmm.. I intentionally suggested to be verbose (to explicitly say the meaning of the boolean), but only when actually worthful (only on positive values) -due to the verbosity-, but I didn't know it cause issues (I still cannot see what's wrong with this approach, honestly).

Then, just ignore my suggestion and merge it as-is! :)

Comment (guicaulada) [reply pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go]: I guess it's not that verbose, verbose is clearer so it's better.

Done on d1e45b8

Comment (joanlopez) [reply pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go]: Great, thanks @guicaulada! 👌🏻

**Pull request merged**